services/shhext/mailservers/connmanager.go

```diff
@@ -148,6 +139,26 @@ func (ps *ConnectionManager) Stop() {
 	ps.quit = nil
 }
 
+func processReplacement(srv PeerAdderRemover, target int, timeout time.Duration, newNodes []*enode.Node, events <-chan *p2p.PeerEvent, connected map[enode.ID]struct{}, current map[enode.ID]*enode.Node) map[enode.ID]*enode.Node {
```

Comment:
Any reason this function is not a method on `ConnectionManager`? We could avoid that many arguments.

Reply:
it doesn't change any internal state of the connection manager, all operations are on local data. i can wrap all local data into a single struct though

Reply:
Yeah, maybe that could help because it's very hard to reason about which parameters are mutable and what's the end result of these operations.